test/testdrive/copy-to-s3-minio.td

```diff
@@ -160,22 +160,22 @@ contains:S3 bucket path is not empty
 $ set-from-sql var=key-1
 SELECT TO_CHAR(now(), 'YYYY-MM-DD')
 
-$ s3-verify-data bucket=copytos3 key=test/1/${key-1}
+$ s3-verify-data bucket=copytos3 key=test/1/${key-1} sort-rows=true
```

Comment:
I think we want to guarantee a sort order in these files. Is the problem that we are currently relying on Row's Ord impl instead of the SQL ordering that one would get in an ORDER BY *?

Reply:
I don't think the S3 sink has any ordering guarantees, so it's reasonable to sort data here.

Reply:
In the [above-linked slack thread](https://materializeinc.slack.com/archives/C01LKF361MZ/p1725905144170379), @rjobanp and Nikhil wrote that the sort order is not guaranteed.

(It's possible that an ORDER BY would make the test pass, but if the ordering is not actually guaranteed by the S3 sink, then it would still be relying on something that is not guaranteed.)

Reply:
Correct - we don't guarantee ordering within the s3 output commands - we just guarantee that partitioning is stable across files (since multiple workers are writing to the same objects)

Reply:
> we just guarantee that partitioning is stable across files (since multiple workers are writing to the same objects

ah, I was misremembering this aspect and thought it also applied to the contents. Thanks the explanations everyone!